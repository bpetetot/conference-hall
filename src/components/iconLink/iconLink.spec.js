/* eslint-env jest */
import snap from 'tests/snapshot'
import IconLink from './index.js'

jest.mock('../iconLabel', () => 'IconLabel')

const snapshot = props => snap(IconLink)({ ...props })

describe('components/iconLink', () => {
  it('should not render if no label', snapshot({ icon: 'i', href: 'h' }))
  it('should not render if no href', snapshot({ icon: 'i', label: 'h' }))
  it('should render icon link', snapshot({ icon: 'i', label: 'l', href: 'h' }))
})
