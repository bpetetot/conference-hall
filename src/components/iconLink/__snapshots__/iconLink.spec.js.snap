// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/iconLink should not render if no href 1`] = `null`;

exports[`components/iconLink should not render if no label 1`] = `null`;

exports[`components/iconLink should render icon link 1`] = `
<a
  className="icon-link"
  href="h"
  rel="noopener noreferrer"
  target="_blank"
>
  <IconLabel
    icon="i"
    label="l"
  />
</a>
`;
